mapie/tests/test_classification.py

```diff
@@ -833,3 +833,36 @@ def test_pred_loof_isnan() -> None:
         k=0,
     )
     assert len(y_pred) == 0
+
+
+@pytest.mark.parametrize("strategy", [*STRATEGIES])
+def test_pipeline_compatibility(strategy: str) -> None:
+    """Check that MAPIE works on pipeline based on pandas dataframes"""
+    X = pd.DataFrame(
+        {
+            "x_cat": ["A", "A", "B", "A", "A", "B"],
+            "x_num": [0, 1, 1, 4, np.nan, 5],
+        }
+    )
+    y = pd.Series([0, 1, 2, 0, 1, 0])
+    numeric_preprocessor = Pipeline(
+        [
+            ("imputer", SimpleImputer(strategy="mean")),
+        ]
+    )
+    categorical_preprocessor = Pipeline(
+        steps=[
+            ("encoding", OneHotEncoder(handle_unknown="ignore"))
+        ]
+    )
+    preprocessor = ColumnTransformer(
+        [
+            ("cat", categorical_preprocessor, ["x_cat"]),
+            ("num", numeric_preprocessor, ["x_num"])
+        ]
+    )
+    pipe = make_pipeline(preprocessor, DecisionTreeClassifier())
+    pipe.fit(X, y)
+    mapie = MapieClassifier(estimator=pipe, **STRATEGIES[strategy][0])
+    mapie.fit(X, y)
+    mapie.predict(X)
```

Comment:
Shall we test this pipeline against MAPIE predictions using a custom numpy array containing the preprocessing ?